### DISCUSSION TITLE
SearchableRelation questions

### DISCUSSION THREAD

**Discussion by WillyReyno** [laravel/nova-issues#4010] [Q&A] [answered]
Hello,

Is it possible to search on a related json column without case sensitivity?
Is it possible to search through nested relationships? (example: Post -> Comment -> Author -> name) 

Thanks for the awesome work!

**Comment by crynobone** [comment, marked as answer]
You should be able to create your own custom search class as it only requires an invokable object, or just use a `Closure`.

**Reply by WillyReyno** [reply]
I hadn't thought about that option, thanks!

Meanwhile, I found solution for both cases : 

**Search through json column without case sensitivity**
I switched my schema columns from `json` to `text` and it now allows me to search without case sensitivity and everything still works fine in my app.

**Searching through nested relationships**
 I used [staudenmeir/eloquent-has-many-deep](https://github.com/staudenmeir/eloquent-has-many-deep) that allows me to create nested relationships on my models. If I take my example above, I now have a `author` relationship method on my `Post` model, which pass through Comment to get them.

In my `Post` Resource I can now simply use `author.name` and search is working perfectly
